### PR TITLE
[SPARK-45807][SQL] Improve ViewCatalog API

### PR DESCRIPTION
**Pull request by nastra**
### What changes were proposed in this pull request?

Follow-up API improvements based on comments from https://github.com/apache/spark/pull/43677

### Why are the changes needed?

Required for DataSourceV2 view support.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

N/A

### Was this patch authored or co-authored using generative AI tooling?

N/A

### CODE REVIEW

**Comment by nastra** [comment]
@cloud-fan can you please review this one? This is related to https://github.com/apache/spark/pull/43677#discussion_r1421886144

File: sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+ * @param sql the SQL text that defines the view
+ * @param currentCatalog the current catalog
+ * @param currentNamespace the current namespace
+ * @param schema the view query output schema
+ * @param queryColumnNames the query column names
+ * @param columnAliases the column aliases
+ * @param columnComments the column comments
+ * @param properties the view properties
+ */
+public record ViewInfo(Identifier ident,
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java:37]: can we use two space indentation?

Comment (nastra) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java:37]: done

Comment (cloud-fan) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java:37]: it's not done... it should be
```
public record ViewInfo(
  Identifier ident,
  ...
```

Comment (cloud-fan) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java:37]: We have the `ColumnDefaultValue` API in ds v2 and it's just plain java class

Comment (nastra) [reply sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ViewInfo.java]: sorry I misunderstood the first comment. Indeed, no need to use `record` here, so I've switched to a plain java class

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by jzhuge** [comment]
Thanks @nastra @cloud-fan @HyukjinKwon !

**Pull request closed without merging**